src/renderers/shared/reconciler/ReactReconciler.js

```diff
@@ -66,7 +66,9 @@ var ReactReconciler = {
   ) {
     var prevElement = internalInstance._currentElement;
 
-    if (nextElement === prevElement && nextElement._owner != null) {
+    if (nextElement === prevElement &&
+        nextElement._owner != null &&
+        Object.keys(context).length === 0) {
```

Comment:
But a shallow compare isn't good enough, because the context value might have changed deeply.